.gitmodules

```diff
@@ -0,0 +1,3 @@
+[submodule "vcpkg"]
+	path = vcpkg
+	url = https://github.com/mixxxdj/vcpkg.git
```

Comment:
I don't think using vcpkg as submodule works well. I have bad experiences using submodules in general.
In this case this will clutter the source tree with a huge number of files, only some users really need. This also interferrs when using git worktree.
Please let's stick to our scripted solution in the tool files and a vcpkg tree outside the mixxx source tree.

Reply:
If you don't need the submodule then simply don't clone it. It is only needed for macOS and Windows.

Our current solution is not optimal to maintain. Often when starting a new build, the cache has expired since the last build. If there are any build failures, it may waste hours and the packages that did successfully build need to be built all over again. Also there are issues with packages not always getting rebuilt after making a change. In my experience that is not an issue in manifest mode because vcpkg hashes the contents of the port for a cache key.

Using manifest mode will also allow us to easily maintain Qt5 and Qt6 builds simultaneously on one branch.

Reply:
It is also easier to test changes to vcpkg packages with local Mixxx builds in manifest mode.

Reply:
Using manifest mode with the submodule also ensures that Windows and macOS developers are using updated dependencies. Currently the buildenv script has to be manually rerun whenever the dependencies are updated.

Reply:
> If you don't need the submodule then simply don't clone it. It is only needed for macOS and Windows.

That's why we have the scripts in the tools directory. Let's stick to this approach and a vcpkg tree outside the mixxx tree.

Reply:
I don't think "I don't like submodules" is a good enough reason to not do this. It is how vcpkg is designed to be used.

Reply:
As vcpkg adds new features they are likely to be unavailable with our legacy setup. This is already the case with pinning versions.

Reply:
I am am not opposing to the solution at a whole. I am against using the git submodul feature for this. 

We can easily replace this with a normal git call. Submodules are quite unhandy, especially if the submodule is only required for some targets. The submodule feature does not work well with git worktrees.
In this case it becomes worse, because of the size of the "installed*" folder which is ignored by git anyway. 
But it is exactly that folder which contains our requirements.  

In addition  directory structure like this will be much more convenient. 

- workspace
-- mixxx_main
-- mixxx_fetaure1
-- mixxx_fetaure2
-- vcpkg_2.4 

If we use a script from the tools directory or dedicated git commands from the CMakeList.txt for downloading or building the vcpkg folders we have much more chance to control the behaviour and inform the contributors.   
This can contain the commit hash in the same way the submodul feature does in addition to the full download, or whatever fits the best.  

We have discussed using submodules for libraries in our lib folder like here: 
https://github.com/mixxxdj/mixxx/pull/2753#issuecomment-655324002

Reply:
This seems like an elaborate overcomplication that would be worse than using a submodule. This is a different use case from using a submodule for vendored libraries.

> The submodule feature does not work well with git worktrees.

What do you mean? I just tested creating a git worktree from the Tenacity repo. The vcpkg submodule was in the worktree but was not automatically initialized by git, which I think is fine. My use cases for git worktree with Mixxx are working on controller mappings and skins in separate branches from C++, so cloning the vcpkg submodule into those worktrees wouldn't be relevant. So what is the issue?

Reply:
I use work trees for feature branches. If I where on MacOs a central vckpg folder would be convenient for this.

Reply:
You could still do that by explicitly setting the VCPKG_ROOT CMake variable.

Reply:
Setting VCPKG_ROOT is hardly different from what you would have to do to compile different Git worktrees on macOS and Windows already. Either you'd need to redownload and extract a >1 GB ZIP file for each worktree with the buildenv script or you'd need to manually set CMAKE_TOOLCHAIN_FILE to one central location.